File: Cpp/435.cpp
#include<iostream>
#include<vector>
#include<algorithm>

using namespace std;

struct Interval{
	int start;
	int end;
	Interval():start(0),end(0){}
	Interval(int s,int e):start(s),end(e){}
};

class Solution{
public:
    int eraseOverlapIntervals(vector<Interval>& intervals){
        if(intervals.size()<=1)
            return 0;
        sort(intervals.begin(),intervals.end(),[](Interval i,Interval j){
            return i.end<j.end;
        });
        int cnt=0;
        Interval temp=intervals[0];
        for(int i=1;i<intervals.size();++i){
            if(intervals[i].start<temp.end)
                cnt++;
            else
                temp=intervals[i];
        }
        return cnt;
    }
};

int main(){
	vector<Interval> intervals={{1,2},{2,3},{3,4},{1,3}};
	vector<Interval> intervals2={{1,2},{1,2},{1,2}};
	vector<Interval> intervals3={{1,2},{2,3}};
	Solution s;
	cout<<s.eraseOverlapIntervals(intervals3)<<endl;

	return 0;
}

File: Python/435.py
class Solution(object):
    def eraseOverlapIntervals(self, intervals):
        """
        :type intervals: List[List[int]]
        :rtype: int
        """
        if len(intervals) < 2:
            return 0
        ins = sorted(intervals, key= lambda x:x[1])
        start, res = ins[0], 0
        for i in range(1, len(ins)):
            if ins[i][0] < start[1]:
                res +=1
            else:
                start = ins[i]
        return res
